File: 1734-Decode/1734.java
class Solution {
  public int[] decode(int[] encoded) {
    int n = encoded.length + 1;
    int total = 0;
    for (int i = 1; i <= n; i++) {
      total ^= i;
    }
    int odd = 0;
    for (int i = 1; i < n - 1; i += 2) {
      odd ^= encoded[i];
    }
    int[] perm = new int[n];
    perm[0] = total ^ odd;
    for (int i = 0; i < n - 1; i++) {
      perm[i + 1] = perm[i] ^ encoded[i];
    }
    return perm;
  }
}

File: 1734-Decode/1734.py
class Solution:
  def decode(self, encoded: List[int]) -> List[int]:
    n = len(encoded) + 1
    total = reduce(xor, range(1, n + 1))
    odd = 0
    for i in range(1, n - 1, 2):
      odd ^= encoded[i]

    perm = [total ^ odd]
    for i in range(n - 1):
      perm.append(perm[-1] ^ encoded[i])

    return perm